pkg/kube/pod_controller.go

```diff
@@ -40,9 +40,9 @@ var (
 type PodController interface {
 	PodName() string
 	Pod() *corev1.Pod
-	StartPod(ctx context.Context, stopTimeout time.Duration) error
+	StartPod(ctx context.Context) error
```

Comment:
Removing `stopTimeout` is good idea. I was thinking would it be a good idea if `StartPod` accepts another parameter that decides if `StartPod` wold also wait for the pod to be Running.
But I see we have another method that does that.
I am not sure if removing `WaitForPodReady` method and having that handled in `StartPod` would be better idea but it's worth thinking about.

Reply:
Originally it was written in a way you are describing, but I think that generic interface should provide granular methods, and consumers are able to combine them as they would like. Also that makes tests easier.

Reply:
Agree with @e-sumin too 👍🏼